pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -90,19 +90,15 @@ func (builder *OnBuild) Build(config *api.Config) (*api.Result, error) {
 	// Change the installation directory for this config to store scripts inside
 	// the application root directory.
 	if err := builder.source.Prepare(config); err != nil {
-		return buildResult, err
+		return nil, err
 	}
 
 	// If necessary, copy the STI scripts into application root directory
 	builder.copySTIScripts(config)
 
 	glog.V(2).Info("Creating application Dockerfile")
 	if err := builder.CreateDockerfile(config); err != nil {
-		buildResult.BuildInfo.FailureReason = utilstatus.NewFailureReason(
-			utilstatus.ReasonDockerfileCreateFailed,
-			utilstatus.ReasonMessageDockerfileCreateFailed,
-		)
-		return buildResult, err
```

Comment:
though we don't support on-build in openshift today, i don't see that as a good reason to remove this code.  what is the harm in having it?